### PR TITLE
chore: Moving to UI generalisation V2

### PR DESCRIPTION
**Pull request by mnorbury**
Migrating Postgres destination to UI V2. This follows the pattern established in the MySQL destination plugin.

For Postgres, we've tried to preserve the usage of key-value pair connection strings when using the regular setup. This allows special characters to be used without escaping (due to the value being encompassed in `'`), in addition it allows the UI to be easily re-populated when the plugin is edited.

refs: https://github.com/cloudquery/cloudquery-issues/issues/2474

### CODE REVIEW

File: plugins/destination/postgresql/cloud-config-ui/src/utils/prepareInitialValues.ts
Comment (mnorbury) [inline plugins/destination/postgresql/cloud-config-ui/src/utils/prepareInitialValues.ts:1]: Maybe we should be calling this in the `useConfig` function?

File: plugins/destination/postgresql/cloud-config-ui/src/App.tsx [hunk 1]
```diff
@@ -1,51 +1,65 @@
...
+  const { initialValues, initialized, teamName, context } = useFormInit(
     pluginUiMessageHandler,
     false,
   );
-  useFormHeightChange(pluginUiMessageHandler);
+  // useFormHeightChange(pluginUiMessageHandler);
 
   const theme = useMemo(() => createTheme(createThemeOptions()), []);
 
+  const config = useConfig({ initialValues });
```
Comment (mnorbury) [inline plugins/destination/postgresql/cloud-config-ui/src/App.tsx:43]: Previously we called `prepareInitialValues` before passing them to the Form to be rendered, maybe we also want to do this here - until we refactor the util functions.

File: plugins/destination/postgresql/cloud-config-ui/e2e-tests/validate-config.spec.ts [hunk 2]
```diff
@@ -4,12 +4,12 @@ import YAML from 'yaml';
 
 test('Submit the form', async ({ page }) => {
   await page.goto('/');
-  await page.getByLabel('Host *').click();
```
Comment (mnorbury) [inline plugins/destination/postgresql/cloud-config-ui/e2e-tests/validate-config.spec.ts:7]: It looks like `*` used to mean required, but now means secret.

Comment (cduflo) [reply plugins/destination/postgresql/cloud-config-ui/e2e-tests/validate-config.spec.ts:7]: yeah, this is on my list too, but added here for visibility: https://github.com/cloudquery/cloudquery-issues/issues/2486

**Review by cduflo — APPROVED** [review]
epic stuff @mnorbury 🚀 

couple small edits then I'm psyched to see this merged! thanks for doing this one, the connection string stuff was definitely tricky, loving the test coverage 👍

File: plugins/destination/postgresql/cloud-config-ui/src/hooks/useConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+                  </>
+                ),
+              },
+            ],
+          },
+          {
+            header: 'Setup guide',
+            bodies: [
+              {
+                text: `To allow CloudQuery network access to your MySQL instance, make sure the following
```
[the remark this answers is not in the document]
Comment (mnorbury) [reply plugins/destination/postgresql/cloud-config-ui/src/hooks/useConfig.tsx]: Fixed in https://github.com/cloudquery/cloudquery/pull/19237/commits/8699b7e73f30c9b17a6917acc565e69bad546252

File: .github/workflows/dest_postgresql.yml [hunk 4]
```diff
@@ -75,13 +75,97 @@ jobs:
...
+          echo "ALIAS_DEPLOYMENT_URL=$ALIAS_DEPLOYMENT_URL" >> $GITHUB_ENV
+
+      - name: Find Comment
+        uses: peter-evans/find-comment@v3
+        id: find-comment
+        with:
+          token: ${{ secrets.GH_CQ_BOT }}
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: "cq-bot"
+          body-includes: "postgresql Cloud UI deployed to Vercel:"
```
[the remark this answers is not in the document]
Comment (mnorbury) [reply .github/workflows/dest_postgresql.yml]: Good catch, this was probably going to eventually break Github :)

Fixed [here](https://github.com/cloudquery/cloudquery/pull/19237/commits/0593a346bf329709d98d3d950e937f8b4adcd71f)

File: plugins/destination/postgresql/cloud-config-ui/src/hooks/useConfig.tsx [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+                CloudQuery IPs are in your firewall allowlist:`,
+              },
+              { code: '35.231.218.115' },
+              { code: '35.231.72.234' },
+            ],
+          },
+        ],
+      },
+      errorCodes: {
```
[the remark this answers is not in the document]
Comment (mnorbury) [reply plugins/destination/postgresql/cloud-config-ui/src/hooks/useConfig.tsx:312]: Aligned with Postgres error code [here](https://github.com/cloudquery/cloudquery/pull/19237/commits/660f4feb52a63606f38eaeeeb1abf9a68128f763)

**Review by cduflo — APPROVED** [review]
LGTM :)

**Pull request merged**